File: P1/Ejercicio3/src/constructor/ConstructorBicicletaCarretera.java
package constructor;

import bicicleta.BicicletaCarretera;

/**
 * Clase constructora para bicicletas de carretera. Esta hereda de Constructor
 */
public class ConstructorBicicletaCarretera extends Constructor {
    /**
     * Constructor de la clase sin parámetros. Inicializa la bicicleta a una bicicleta de carretera por defecto
     */
    public ConstructorBicicletaCarretera() { bicicleta = new BicicletaCarretera(); }

    /**
     * Implementación del método abstracto 'recomponer', que reinicializa la instancia de bicicleta de carretera
     */
    @Override
    public void recomponer() {
        bicicleta = new BicicletaCarretera();
    }
}

File: P2/lib/Constructores/constructorBicicletaCarretera.dart
import 'constructor.dart';
import '../Bicicleta/BicicletaCarretera.dart';

class ConstructorBicicletaCarretera extends Constructor {

  ConstructorBicicletaCarretera() {
    bicicleta = BicicletaCarretera();
  }

  @override
  void recomponer() {
    bicicleta = BicicletaCarretera();
  }
}
